### ISSUE TITLE
Bug: Images not moving by drag and drop inside the editor

### ISSUE THREAD

**Issue by PauloMendees** [facebook/lexical#6691]
Lexical version: 0.12.5

## Steps To Reproduce

1. Create an editor following the documentation (using react)
2. Copy the ImageNode and ImageComponent from the playground
3. Add/install the necessary dependencies of ImageComponent and ImageNode
4. Run the code

## The current behavior
https://www.loom.com/share/61ac1ec806e14e14a321a7a30f217f0d?sid=f80b5254-5542-4d7e-afdb-9415cbe6de5f

## The expected behavior
Be able to move images by draggind and drop

**Comment by moy2010** [comment]
The code found in the playground is not expected to be production-ready, and definitely it's not expected to cover all use cases. It's only provided as example code.

**Comment by etrepum** [comment]
Yeah I just don't think that this is implemented at all. I believe we have drag and drop to insert a new image, but not to move an existing one. A PR to implement this would be appreciated!

**Comment by vantage-ola** [comment]
hi @etrepum  from the mlh fellowship fall 24' program... 
I would love to work on this issue. can you assign me?
@PauloMendees are you working on this? so I can go ahead

**Comment by etrepum** [comment]
Assignment isn't needed to work on an issue, when you have a solution just submit a pull request

**Comment by vantage-ola** [comment]
this is my first issue I and my mates are going to work  on... from my understanding, can we take the approach of  writing a plugin interface or working directly on the lexical core pkg.

Can you pls give me some advice on how to approach this issue, thanks

**Comment by vantage-ola** [comment]
looking at the lexical-playground ... 

 this for moving the img around
```
      editor.registerCommand<DragEvent>(
        DRAGOVER_COMMAND,
        (event) => {
          return $onDragover(event);
        },
        COMMAND_PRIORITY_LOW,
      )
```

```
function canDropImage(event: DragEvent): boolean {
  const target = event.target;
  return !!(
    target &&
    target instanceof HTMLElement &&
    !target.closest('code, span.editor-image') &&
    target.parentElement &&
    target.parentElement.closest('div.ContentEditable__root')
  );
}
```

```
function $onDragover(event: DragEvent): boolean {
  const node = $getImageNodeInSelection();
  if (!node) {
    return false;
  }
  if (!canDropImage(event)) {
    event.preventDefault();
  }
  return true;
}

```

the `DRAG_OVER_COMMAND` already exists in `lexical` .... how do you suggest i make it a feature

**Comment by etrepum** [comment]
The image code is entirely within the playground, so you don't have to worry much about making it too generic or reusable outside of that context.

You can follow the [Contributing](https://github.com/facebook/lexical/blob/main/CONTRIBUTING.md) guide to get set up. In this case the relevant code for image nodes and drag and drop is entirely within the playground, so you don't have to worry so much about other use cases or documentation. I don't have any expertise with this specific code so I couldn't tell you exactly what needs to be done to make it work, but I would get started by doing a careful read-through of the code and the [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API).

**Comment by LeSingh1** [comment]
I can't reproduce this in the playground on current `main`. Moving an image by drag and drop works: click the image once to select it, then drag — it is removed from its old position and inserted at the drop caret. Verified in Chromium and WebKit via `Images.spec.mjs` (10/10 passing, including "Can be dragged and dropped correctly when the image is clicked"), plus ad-hoc drags backwards across paragraphs, into a list item, and to the end of a later paragraph.

@etrepum — one correction: this has been implemented since #2027 (`2ace3db99`, May 2022), and `git show v0.12.5:packages/lexical-playground/src/plugins/ImagesPlugin/index.tsx` confirms it was already present in the version reported here. The catch is *where* it lives: the `DRAGSTART_COMMAND` / `DRAGOVER_COMMAND` / `DROP_COMMAND` handlers are in the images plugin (`plugins/ImagesExtension/index.tsx`, formerly `plugins/ImagesPlugin`), not in `ImageNode` or `ImageComponent`.

@PauloMendees — your steps say you copied `ImageNode` and `ImageComponent`. Copying `ImagesPlugin` as well and registering it should give you drag-to-move.

One real limitation, which may be what the video shows: you must click the image to select it before dragging. `ImageComponent` calls `preventDefault()` on `dragstart` and only sets `draggable` once the node is selected. That is intentional and pinned by the existing test "Cannot be dragged without being clicked" — dragging an unselected image would be a separate feature request.
